crates/sui-rpc-api/proto/sui.types.proto

```diff
@@ -0,0 +1,1040 @@
+syntax = "proto3";
```

Comment:
maybe this is annoying for me to ask at this point, but I wonder if there's a way this file can be generated completely mechanically by some annotations/build script/etc. on the source types in rust? that would have the benefits that it could never get out of sync, all the field names would match exactly, the source of everything would always be 100% clear, etc.

(maybe you'd need to keep some of the primitives like Address, Digest, etc. manually but then ideally all the more-complicated types could be generated)

Reply:
That's something that we could do, but I think it would require a great deal of infra to be written and designed just for this purpose. At this point I think doing that work would take a number of extra weeks of work (which may not be worth it at this point since it would largely result in us having exactly what we have here)

Reply:
My instinct says it would be a very worthwhile investment if we need this to stay in sync with sui core types as they evolve and would save a lot more time than it costs,

AND also that if you need to defer this work for pragmatic reasons then you are the best person to decide that!